RecoTracker/FinalTrackSelectors/plugins/TrackCutClassifier.cc

```diff
@@ -93,6 +93,33 @@ namespace {
     }
   }
   
+  inline void dzCut_wPVerror_par(reco::Track const & trk, int & nLayers, const float * par, const int * exp, Point const & bestVertexError, float dzCut[]) {
+    float dzE = trk.dzError();
+    float zPVerr = bestVertexError.z();
+
+    float dzErrPV = std::sqrt(dzE*dzE+zPVerr*zPVerr);
+    for (int i=2; i>=0; --i) {
+      if (exp[i] != 0)
+	dzCut[i] = par[i]*pow(nLayers,exp[i])*dzErrPV; 
+      else
+	dzCut[i] = par[i]*dzErrPV; 	
+    }
+  }
+
+  inline void drCut_wPVerror_par(reco::Track const & trk, int & nLayers, const float* par, const int * exp, Point const & bestVertexError, float drCut[]) {
+    float drE = trk.d0Error();
+    float rPVerr = sqrt(bestVertexError.x()*bestVertexError.y()); // shouldn't it be bestVertex.xError()*bestVertex.xError()+bestVertex.yError()*bestVertex.yError() ?!?!?
+
+    float drErrPV = std::sqrt(drE*drE+rPVerr*rPVerr);
+    for (int i=2; i>=0; --i) {
+      if (exp[i] != 0)
+	drCut[i] = par[i]*pow(nLayers,exp[i])*drErrPV;
+      else
+	drCut[i] = par[i]*drErrPV;
```

Comment:
The `if` statement would probably be better written as:
```
drCut[i] = par[i]*drErrPV;
if (exp[i] != 0)
	drCut[i] *= pow(nLayers,exp[i]);
```
Same for lines 102-105.